{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "#install conda install -c conda-forge scikit-allel\n",
    "import pandas as pd\n",
    "import allel\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "samples=['AKR_J','A_J','BALB_cJ','C3H_HeJ','CBA_J','DBA_2J','LP_J']\n",
    "# tools=['sniffles']\n",
    "tools=['breakdancer',\n",
    "#'clever',\n",
    "'delly',\n",
    "'gasv',\n",
    "'gridss',\n",
    "#'indelminer',\n",
    "#'lumpexpress',\n",
    "#'mistrvar',\n",
    "#'pindel',\n",
    "#'platypus',\n",
    "'popdel',\n",
    "'smoove',\n",
    "'crest',\n",
    "'genomestrip',\n",
    "'manta_diploidSV'\n",
    "#'sniffles'\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- AKR_J ---\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vs/.local/lib/python3.5/site-packages/ipykernel_launcher.py:36: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- A_J ---\n",
      "--- BALB_cJ ---\n",
      "--- C3H_HeJ ---\n",
      "--- CBA_J ---\n",
      "--- DBA_2J ---\n",
      "--- LP_J ---\n"
     ]
    }
   ],
   "source": [
    "#tools\n",
    "import os\n",
    "from os import path\n",
    "\n",
    "#cov_list=['32','8','4','2','1','0.5','0.1']\n",
    "#cov_list=['32','8','4','2']\n",
    "cov_list=['32','16','8','4','2','1','0.5','0.1']\n",
    "th_list=['0','10','100','1000','10000']\n",
    "n_list=['1','2','3','4','5','6','7','8','9','10']\n",
    "\n",
    "\n",
    "#n_list=['1']\n",
    "\n",
    "\n",
    "df= pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "\n",
    "\n",
    "\n",
    "for s in samples:\n",
    "    print (\"---\",s,\"---\")\n",
    "    for t in tools:\n",
    "        for th in th_list: \n",
    "            for cov in cov_list:\n",
    "                for n in n_list:\n",
    "            \n",
    "            \n",
    "                    file='../Data/raw_data/mouse/custom_vcf_'+cov+'x/'+str(th)+'t/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.'+cov+'p.'+n+'_sorted.modified.vcf'\n",
    "            \n",
    "                    if path.exists(file):\n",
    "                        callset = allel.read_vcf(file,fields='*')\n",
    "                        if callset!=None:\n",
    "                            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                            df_current['cov']=cov\n",
    "                            df_current['n']=n\n",
    "                            df = pd.concat([df_current, df],ignore_index=True) \n",
    "#                         else:\n",
    "#                             print(\"Empty File: \" + file)\n",
    "\n",
    "#                     else:\n",
    "#                         print(\"Nonexistant File: \" + file)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- AKR_J ---\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vs/.local/lib/python3.5/site-packages/ipykernel_launcher.py:36: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.AKR_J.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.AKR_J.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.AKR_J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.AKR_J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.AKR_J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.AKR_J.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.AKR_J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.AKR_J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.AKR_J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.AKR_J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.AKR_J.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.AKR_J.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.AKR_J.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.AKR_J.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.AKR_J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.AKR_J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.AKR_J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.AKR_J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.AKR_J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.AKR_J.chr19.0.5p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.AKR_J.chr19.0.5p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.AKR_J.chr19.0.5p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.AKR_J.chr19.0.5p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.AKR_J.chr19.0.5p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.AKR_J.chr19.0.5p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.AKR_J.chr19.0.5p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.AKR_J.chr19.0.5p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.AKR_J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.AKR_J.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.AKR_J.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.AKR_J.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.AKR_J.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.AKR_J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.AKR_J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.AKR_J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.AKR_J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.AKR_J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "--- A_J ---\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.A_J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.A_J.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.A_J.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.A_J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.A_J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.A_J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.A_J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.A_J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.A_J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.A_J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.A_J.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.A_J.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.A_J.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.A_J.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.A_J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.A_J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.A_J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.A_J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.A_J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.A_J.chr19.0.5p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.A_J.chr19.0.5p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.A_J.chr19.0.5p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.A_J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.A_J.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.A_J.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.A_J.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.A_J.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.A_J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.A_J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.A_J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.A_J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.A_J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "--- BALB_cJ ---\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.BALB_cJ.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.BALB_cJ.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.BALB_cJ.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.BALB_cJ.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.BALB_cJ.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.BALB_cJ.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.BALB_cJ.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.BALB_cJ.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.BALB_cJ.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.BALB_cJ.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.BALB_cJ.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.BALB_cJ.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.BALB_cJ.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.BALB_cJ.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.BALB_cJ.chr19.0.1p.10_sorted.modified.nondel.vcf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.BALB_cJ.chr19.0.5p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.BALB_cJ.chr19.0.5p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.BALB_cJ.chr19.0.5p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.BALB_cJ.chr19.0.5p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.BALB_cJ.chr19.0.5p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.BALB_cJ.chr19.0.5p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.BALB_cJ.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.BALB_cJ.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.BALB_cJ.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.BALB_cJ.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.BALB_cJ.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.BALB_cJ.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.BALB_cJ.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.BALB_cJ.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.BALB_cJ.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.BALB_cJ.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_16x/100t/nf_100t.genomestrip.BALB_cJ.chr19.16p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_4x/100t/nf_100t.genomestrip.BALB_cJ.chr19.4p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_4x/100t/nf_100t.genomestrip.BALB_cJ.chr19.4p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_4x/100t/nf_100t.genomestrip.BALB_cJ.chr19.4p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_4x/100t/nf_100t.genomestrip.BALB_cJ.chr19.4p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_1x/100t/nf_100t.genomestrip.BALB_cJ.chr19.1p.10_sorted.modified.nondel.vcf\n",
      "--- C3H_HeJ ---\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.C3H_HeJ.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.C3H_HeJ.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.C3H_HeJ.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.C3H_HeJ.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.C3H_HeJ.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.C3H_HeJ.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.C3H_HeJ.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.C3H_HeJ.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.C3H_HeJ.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.C3H_HeJ.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.C3H_HeJ.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.C3H_HeJ.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.C3H_HeJ.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.C3H_HeJ.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.C3H_HeJ.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.C3H_HeJ.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.C3H_HeJ.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.C3H_HeJ.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.C3H_HeJ.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.C3H_HeJ.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.C3H_HeJ.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.C3H_HeJ.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.C3H_HeJ.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.C3H_HeJ.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.C3H_HeJ.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.C3H_HeJ.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.C3H_HeJ.chr19.0.5p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.C3H_HeJ.chr19.0.5p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.C3H_HeJ.chr19.0.5p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.C3H_HeJ.chr19.0.5p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.C3H_HeJ.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.C3H_HeJ.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.C3H_HeJ.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.C3H_HeJ.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.C3H_HeJ.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.C3H_HeJ.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.C3H_HeJ.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.C3H_HeJ.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.C3H_HeJ.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.C3H_HeJ.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_16x/100t/nf_100t.genomestrip.C3H_HeJ.chr19.16p.2_sorted.modified.nondel.vcf\n",
      "--- CBA_J ---\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.CBA_J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.CBA_J.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.CBA_J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.CBA_J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.CBA_J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.CBA_J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.CBA_J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.CBA_J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.CBA_J.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.CBA_J.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.CBA_J.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.CBA_J.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.CBA_J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.CBA_J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.CBA_J.chr19.0.1p.9_sorted.modified.nondel.vcf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.CBA_J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.CBA_J.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.CBA_J.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.CBA_J.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.CBA_J.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.CBA_J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.CBA_J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.CBA_J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.CBA_J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.CBA_J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.CBA_J.chr19.0.5p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.CBA_J.chr19.0.5p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.CBA_J.chr19.0.5p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.CBA_J.chr19.0.5p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.CBA_J.chr19.0.5p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.CBA_J.chr19.0.5p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.CBA_J.chr19.0.5p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.CBA_J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.CBA_J.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.CBA_J.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.CBA_J.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.CBA_J.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.CBA_J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.CBA_J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.CBA_J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.CBA_J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.CBA_J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "--- DBA_2J ---\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.DBA_2J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.DBA_2J.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.DBA_2J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.DBA_2J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.DBA_2J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.DBA_2J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.DBA_2J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.DBA_2J.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.DBA_2J.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.DBA_2J.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.DBA_2J.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.DBA_2J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.DBA_2J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.DBA_2J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.DBA_2J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.DBA_2J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.DBA_2J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.DBA_2J.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.DBA_2J.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.DBA_2J.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.DBA_2J.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.DBA_2J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.DBA_2J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.DBA_2J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.DBA_2J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.DBA_2J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.DBA_2J.chr19.0.5p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.DBA_2J.chr19.0.5p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.DBA_2J.chr19.0.5p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.DBA_2J.chr19.0.5p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.DBA_2J.chr19.0.5p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.DBA_2J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.DBA_2J.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.DBA_2J.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.DBA_2J.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.DBA_2J.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.DBA_2J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.DBA_2J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.DBA_2J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.DBA_2J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.DBA_2J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.genomestrip.DBA_2J.chr19.0.5p.3_sorted.modified.nondel.vcf\n",
      "--- LP_J ---\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.LP_J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.LP_J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.LP_J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.LP_J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.breakdancer.LP_J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.LP_J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.LP_J.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.LP_J.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.LP_J.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.LP_J.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.LP_J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.LP_J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.LP_J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.LP_J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.delly.LP_J.chr19.0.1p.10_sorted.modified.nondel.vcf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.LP_J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.LP_J.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.LP_J.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.LP_J.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.LP_J.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.LP_J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.LP_J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.LP_J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.LP_J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.popdel.LP_J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.LP_J.chr19.0.5p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.LP_J.chr19.0.5p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.LP_J.chr19.0.5p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.LP_J.chr19.0.5p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.LP_J.chr19.0.5p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.LP_J.chr19.0.5p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.5x/100t/nf_100t.smoove.LP_J.chr19.0.5p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.LP_J.chr19.0.1p.1_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.LP_J.chr19.0.1p.2_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.LP_J.chr19.0.1p.3_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.LP_J.chr19.0.1p.4_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.LP_J.chr19.0.1p.5_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.LP_J.chr19.0.1p.6_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.LP_J.chr19.0.1p.7_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.LP_J.chr19.0.1p.8_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.LP_J.chr19.0.1p.9_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_0.1x/100t/nf_100t.smoove.LP_J.chr19.0.1p.10_sorted.modified.nondel.vcf\n",
      "../Data/raw_data/mouse/custom_vcf_16x/100t/nf_100t.genomestrip.LP_J.chr19.16p.6_sorted.modified.nondel.vcf\n"
     ]
    }
   ],
   "source": [
    "#tools\n",
    "import os\n",
    "from os import path\n",
    "\n",
    "# cov_list=['32','8','4','2','1','0.5','0.1']\n",
    "\n",
    "cov_list=['32','16','8','4','2','1','0.5','0.1']\n",
    "th_list=['0','10','100','1000','10000']\n",
    "#th_list=['1000']\n",
    "n_list=['1','2','3','4','5','6','7','8','9','10']\n",
    "\n",
    "\n",
    "#n_list=['1','2']\n",
    "\n",
    "\n",
    "df_nondel= pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "\n",
    "\n",
    "\n",
    "for s in samples:\n",
    "    print (\"---\",s,\"---\")\n",
    "    for t in tools:\n",
    "        for th in th_list: \n",
    "            for cov in cov_list:\n",
    "                for n in n_list:\n",
    "            \n",
    "            \n",
    "                    file='../Data/raw_data/mouse/custom_vcf_'+cov+'x/'+str(th)+'t/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.'+cov+'p.'+n+'_sorted.modified.nondel.vcf'\n",
    "            \n",
    "                    if path.exists(file):\n",
    "                        callset = allel.read_vcf(file,fields='*')\n",
    "                        if callset!=None:\n",
    "                            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                            df_current['cov']=cov\n",
    "                            df_current['n']=n\n",
    "                            df_nondel = pd.concat([df_current, df_nondel],ignore_index=True)   \n",
    "                    else:\n",
    "                        print (file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cov</th>\n",
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>n</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.1</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>manta_diploidSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.1</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>manta_diploidSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.1</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>manta_diploidSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.1</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>manta_diploidSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.1</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>manta_diploidSV</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   cov flag length   n position strain threshold             tool\n",
       "0  0.1          -1  10        0   LP_J       100  manta_diploidSV\n",
       "1  0.1          -1  10        0   LP_J       100  manta_diploidSV\n",
       "2  0.1          -1   9        0   LP_J       100  manta_diploidSV\n",
       "3  0.1          -1   9        0   LP_J       100  manta_diploidSV\n",
       "4  0.1          -1   8        0   LP_J       100  manta_diploidSV"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_nondel.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# group by TP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>n</th>\n",
       "      <th>cov</th>\n",
       "      <th>nTP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [tool, threshold, strain, n, cov, nTP]\n",
       "Index: []"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_data_TP=df[df['flag'] == 'TP'].groupby(['tool','threshold','strain','n','cov'],as_index=False)['flag'].count()\n",
    "group_data_TP=group_data_TP.rename(columns={\"flag\": \"nTP\"})\n",
    "\n",
    "\n",
    "group_data_TP[pd.isnull(group_data_TP).any(axis=1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>n</th>\n",
       "      <th>cov</th>\n",
       "      <th>nFP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [tool, threshold, strain, n, cov, nFP]\n",
       "Index: []"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_data_FP=df[df['flag'] == 'FP'].groupby(['tool','threshold','strain','n','cov'],as_index=False)['flag'].count()\n",
    "group_data_FP=group_data_FP.rename(columns={\"flag\": \"nFP\"})\n",
    "group_data_FP.tail()\n",
    "\n",
    "\n",
    "group_data_FP[pd.isnull(group_data_FP).any(axis=1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>n</th>\n",
       "      <th>cov</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [tool, threshold, strain, n, cov, nTN]\n",
       "Index: []"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_data_TN=df_nondel[df_nondel['flag'] == 'TN'].groupby(['tool','threshold','strain','n','cov'],as_index=False)['flag'].count()\n",
    "group_data_TN=group_data_TN.rename(columns={\"flag\": \"nTN\"})\n",
    "\n",
    "\n",
    "group_data_TN[pd.isnull(group_data_TN).any(axis=1)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>n</th>\n",
       "      <th>cov</th>\n",
       "      <th>nTP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [tool, threshold, strain, n, cov, nTP]\n",
       "Index: []"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#fill mssing combinations with 0s\n",
    "for s in samples:\n",
    "    for t in tools:\n",
    "        for th in th_list:\n",
    "            for cov in cov_list:\n",
    "                for n in n_list:\n",
    "                    if not (((group_data_TP['cov'] == cov) & (group_data_TP['n'] == n) & (group_data_TP['tool'] == t) & (group_data_TP['strain'] == s) & (group_data_TP['threshold'] == th)).any()):\n",
    "                        group_data_TP = group_data_TP.append({'cov' : cov , 'n' : n , 'tool' : t , 'strain' : s,'threshold' : th,'nTP' : 0.0} , ignore_index=True)\n",
    "group_data_TP[pd.isnull(group_data_TP).any(axis=1)]\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>n</th>\n",
       "      <th>cov</th>\n",
       "      <th>nFP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [tool, threshold, strain, n, cov, nFP]\n",
       "Index: []"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#fill mssing combinations with 0s\n",
    "for s in samples:\n",
    "    for t in tools:\n",
    "        for th in th_list:\n",
    "            for cov in cov_list:\n",
    "                for n in n_list:\n",
    "                    if not (((group_data_FP['cov'] == cov) & (group_data_FP['n'] == n) & (group_data_FP['tool'] == t) & (group_data_FP['strain'] == s) & (group_data_FP['threshold'] == th)).any()):\n",
    "                        group_data_FP = group_data_FP.append({'cov' : cov , 'n' : n , 'tool' : t , 'strain' : s,'threshold' : th,'nFP' : 0.0} , ignore_index=True)\n",
    "group_data_FP[pd.isnull(group_data_FP).any(axis=1)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>n</th>\n",
       "      <th>cov</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [tool, threshold, strain, n, cov, nTN]\n",
       "Index: []"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#fill mssing combinations with 0s\n",
    "for s in samples:\n",
    "    for t in tools:\n",
    "        for th in th_list:\n",
    "            for cov in cov_list:\n",
    "                for n in n_list:\n",
    "                    if not (((group_data_TN['cov'] == cov) & (group_data_TN['n'] == n) & (group_data_TN['tool'] == t) & (group_data_TN['strain'] == s) & (group_data_TN['threshold'] == th)).any()):\n",
    "                        group_data_TN = group_data_TN.append({'cov' : cov , 'n' : n , 'tool' : t , 'strain' : s,'threshold' : th,'nTN' : 0.0} , ignore_index=True)\n",
    "group_data_TN[pd.isnull(group_data_TN).any(axis=1)]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>n</th>\n",
       "      <th>cov</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>63.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "      <td>32</td>\n",
       "      <td>106.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>49.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          tool threshold strain  n cov    nTN\n",
       "0  breakdancer       100  AKR_J  1  16   63.0\n",
       "1  breakdancer       100  AKR_J  1   2    4.0\n",
       "2  breakdancer       100  AKR_J  1  32  106.0\n",
       "3  breakdancer       100  AKR_J  1   4   29.0\n",
       "4  breakdancer       100  AKR_J  1   8   49.0"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_data_TN.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>n</th>\n",
       "      <th>cov</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [tool, threshold, strain, n, cov, nTP, nFP, nTN]\n",
       "Index: []"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#now both TP and FP are the size size and we can merge\n",
    "df_merge=pd.merge(group_data_TP, group_data_FP,on=['tool','threshold','strain','n','cov'])\n",
    "df_merge=pd.merge(df_merge, group_data_TN,on=['tool','threshold','strain','n','cov'])\n",
    "df_merge[pd.isnull(df_merge).any(axis=1)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>n</th>\n",
       "      <th>cov</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "      <td>0.5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>14.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>176.0</td>\n",
       "      <td>111.0</td>\n",
       "      <td>63.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>42.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "      <td>32</td>\n",
       "      <td>217.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>106.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          tool threshold strain  n  cov    nTP    nFP    nTN\n",
       "0  breakdancer       100  AKR_J  1  0.5    3.0    1.0    0.0\n",
       "1  breakdancer       100  AKR_J  1    1   14.0   11.0    0.0\n",
       "2  breakdancer       100  AKR_J  1   16  176.0  111.0   63.0\n",
       "3  breakdancer       100  AKR_J  1    2   42.0   28.0    4.0\n",
       "4  breakdancer       100  AKR_J  1   32  217.0  169.0  106.0"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vs/.local/lib/python3.5/site-packages/ipykernel_launcher.py:11: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  # This is added back by InteractiveShellApp.init_path()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain</th>\n",
       "      <th>n_true</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKR_J</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A_J</td>\n",
       "      <td>533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BALB_cJ</td>\n",
       "      <td>545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBA_J</td>\n",
       "      <td>586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>LP_J</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    strain  n_true\n",
       "0    AKR_J     504\n",
       "1      A_J     533\n",
       "2  BALB_cJ     545\n",
       "3  C3H_HeJ     539\n",
       "4    CBA_J     586\n",
       "5   DBA_2J     609\n",
       "6     LP_J     483"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#true SVs\n",
    "df_current = pd.DataFrame(columns=['strain','length'])\n",
    "df_true = pd.DataFrame(columns=['strain','length'])\n",
    "\n",
    "\n",
    "for s in samples:\n",
    "    file='../Data/gold_standard/mouse_vcf/'+s+'_reference.vcf'\n",
    "    callset = allel.read_vcf(file,fields='*')\n",
    "    \n",
    "    df_current = pd.DataFrame({'strain': s, 'length': callset['variants/SVLEN']})\n",
    "    df_true = pd.concat([df_current, df_true],ignore_index=True)\n",
    "group_data_true = df_true.groupby(['strain'],as_index=False).count()\n",
    "group_data_true=group_data_true.rename(columns={\"length\": \"n_true\"})\n",
    "group_data_true"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>n</th>\n",
       "      <th>cov</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "      <th>n_true</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "      <td>0.5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>14.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>176.0</td>\n",
       "      <td>111.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>42.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "      <td>32</td>\n",
       "      <td>217.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>106.0</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          tool threshold strain  n  cov    nTP    nFP    nTN  n_true\n",
       "0  breakdancer       100  AKR_J  1  0.5    3.0    1.0    0.0     504\n",
       "1  breakdancer       100  AKR_J  1    1   14.0   11.0    0.0     504\n",
       "2  breakdancer       100  AKR_J  1   16  176.0  111.0   63.0     504\n",
       "3  breakdancer       100  AKR_J  1    2   42.0   28.0    4.0     504\n",
       "4  breakdancer       100  AKR_J  1   32  217.0  169.0  106.0     504"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge=pd.merge(df_merge, group_data_true)\n",
    "df_merge.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_merge.to_csv(r'FigureS10_coverage.csv')\n",
    "# df_merge = pd.read_csv('Figure3_df_merge.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
