{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import allel\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "strain=['A_J','AKR_J','BALB_CJ','CBA_J','C3H_HeJ','DBA_2J','LP_J']\n",
    "\n",
    "tools=['BioGraph*',\n",
    "'breakdancer',\n",
    "'clever',\n",
    "'delly',\n",
    "'gasv',\n",
    "'gridss',\n",
    "'indelminer',\n",
    "'lumpexpress',\n",
    "'mistrvar',\n",
    "'pindel',\n",
    "'popdel',\n",
    "'rdxplorer',\n",
    "'smoove',\n",
    "'sniffles']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "%store -r\n",
    "pdList = [df_10,df_100,df_1000,df_10000,df_10_nondel,df_100_nondel,df_1000_nondel,df_10000_nondel] \n",
    "df_cat = pd.concat(pdList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1725</td>\n",
       "      <td>FP</td>\n",
       "      <td>6055203</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>34582728</td>\n",
       "      <td>FP</td>\n",
       "      <td>17098523</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>48</td>\n",
       "      <td>FP</td>\n",
       "      <td>17201451</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>114</td>\n",
       "      <td>FP</td>\n",
       "      <td>17231867</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>35</td>\n",
       "      <td>FP</td>\n",
       "      <td>17291187</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       tool strain    length flag  position threshold\n",
       "0  sniffles   LP_J      1725   FP   6055203        10\n",
       "1  sniffles   LP_J  34582728   FP  17098523        10\n",
       "2  sniffles   LP_J        48   FP  17201451        10\n",
       "3  sniffles   LP_J       114   FP  17231867        10\n",
       "4  sniffles   LP_J        35   FP  17291187        10"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cat.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 50-100 bp Deletions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>68</td>\n",
       "      <td>FP</td>\n",
       "      <td>22431473</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>73</td>\n",
       "      <td>FP</td>\n",
       "      <td>23210349</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>52</td>\n",
       "      <td>FP</td>\n",
       "      <td>25674538</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>52</td>\n",
       "      <td>FP</td>\n",
       "      <td>27784186</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>84</td>\n",
       "      <td>FP</td>\n",
       "      <td>29442320</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool strain length flag  position threshold\n",
       "23  sniffles   LP_J     68   FP  22431473        10\n",
       "27  sniffles   LP_J     73   FP  23210349        10\n",
       "32  sniffles   LP_J     52   FP  25674538        10\n",
       "37  sniffles   LP_J     52   FP  27784186        10\n",
       "41  sniffles   LP_J     84   FP  29442320        10"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cat_50_100=df_cat.loc[(df_cat['length'] >= 50) & (df_cat['length'] <100)]\n",
    "df_cat_50_100.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTP\n",
       "0  BioGraph*         10    AKR_J    2\n",
       "1  BioGraph*         10      A_J    2\n",
       "2  BioGraph*         10  BALB_CJ    1\n",
       "3  BioGraph*         10  C3H_HeJ    2\n",
       "4  BioGraph*         10    CBA_J    3"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_TP_50_100 = df_cat_50_100[df_cat_50_100['flag'] == 'TP'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_TP_50_100 = group_TP_50_100.rename(columns={\"flag\": \"nTP\"})\n",
    "group_TP_50_100.shape\n",
    "group_TP_50_100.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nFP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nFP\n",
       "0  BioGraph*         10    AKR_J  132\n",
       "1  BioGraph*         10      A_J  136\n",
       "2  BioGraph*         10  BALB_CJ  145\n",
       "3  BioGraph*         10  C3H_HeJ  153\n",
       "4  BioGraph*         10    CBA_J  169"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_FP_50_100=df_cat_50_100[df_cat_50_100['flag'] == 'FP'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_FP_50_100=group_FP_50_100.rename(columns={\"flag\": \"nFP\"})\n",
    "group_FP_50_100.shape\n",
    "group_FP_50_100.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>delly</td>\n",
       "      <td>1000</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>delly</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>delly</td>\n",
       "      <td>10000</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>gasv</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>gasv</td>\n",
       "      <td>10000</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    tool  threshold   strain  nTN\n",
       "0  delly       1000  BALB_CJ    1\n",
       "1  delly      10000    AKR_J    1\n",
       "2  delly      10000  BALB_CJ    1\n",
       "3   gasv      10000    AKR_J    3\n",
       "4   gasv      10000      A_J    1"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_TN_50_100=df_cat_50_100[df_cat_50_100['flag'] == 'TN'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_TN_50_100=group_TN_50_100.rename(columns={\"flag\": \"nTN\"})\n",
    "group_TN_50_100.shape\n",
    "group_TN_50_100.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 4)"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_TP_50_100['tool'] == t) & (group_TP_50_100['strain'] == s) & (group_TP_50_100['threshold'] == th)).any()):\n",
    "                group_TP_50_100 = group_TP_50_100.append({'tool' : t , 'strain' : s,'threshold' : th,'nTP' : 0} , ignore_index=True)\n",
    "group_TP_50_100.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 4)"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_FP_50_100['tool'] == t) & (group_FP_50_100['strain'] == s) & (group_FP_50_100['threshold'] == th)).any()):\n",
    "                group_FP_50_100 = group_FP_50_100.append({'tool' : t , 'strain' : s,'threshold' : th,'nFP' : 0} , ignore_index=True)\n",
    "                #print (s,t,th)\n",
    "group_FP_50_100.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 4)"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_TN_50_100['tool'] == t) & (group_TN_50_100['strain'] == s) & (group_TN_50_100['threshold'] == th)).any()):\n",
    "                group_TN_50_100 = group_TN_50_100.append({'tool' : t , 'strain' : s,'threshold' : th,'nTN' : 0} , ignore_index=True)\n",
    "                #print (s,t,th)\n",
    "group_TN_50_100.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>2</td>\n",
       "      <td>132</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>2</td>\n",
       "      <td>136</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>1</td>\n",
       "      <td>145</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>2</td>\n",
       "      <td>153</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>3</td>\n",
       "      <td>169</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>3</td>\n",
       "      <td>164</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>2</td>\n",
       "      <td>124</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>18</td>\n",
       "      <td>116</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>A_J</td>\n",
       "      <td>19</td>\n",
       "      <td>119</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>19</td>\n",
       "      <td>127</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTP  nFP  nTN\n",
       "0  BioGraph*         10    AKR_J    2  132    0\n",
       "1  BioGraph*         10      A_J    2  136    0\n",
       "2  BioGraph*         10  BALB_CJ    1  145    0\n",
       "3  BioGraph*         10  C3H_HeJ    2  153    0\n",
       "4  BioGraph*         10    CBA_J    3  169    0\n",
       "5  BioGraph*         10   DBA_2J    3  164    0\n",
       "6  BioGraph*         10     LP_J    2  124    0\n",
       "7  BioGraph*        100    AKR_J   18  116    0\n",
       "8  BioGraph*        100      A_J   19  119    0\n",
       "9  BioGraph*        100  BALB_CJ   19  127    0"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge_50_100=pd.merge(group_TP_50_100, group_FP_50_100,on=['tool','threshold','strain'])\n",
    "df_merge_50_100=pd.merge(df_merge_50_100, group_TN_50_100,on=['tool','threshold','strain'])\n",
    "df_merge_50_100.shape\n",
    "df_merge_50_100.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_current = pd.DataFrame(columns=['strain','length'])\n",
    "#df = pd.DataFrame(columns=['strain','length'])\n",
    "\n",
    "\n",
    "#for s in strain:\n",
    "    #file='../gold_standard/mouse_vcf/'+s+'_reference.vcf'\n",
    "    #callset = allel.read_vcf(file,fields='*')\n",
    "    \n",
    "    #df_current = pd.DataFrame({'strain': s, 'length': callset['variants/SVLEN']})\n",
    "    #df = pd.concat([df_current, df],ignore_index=True)\n",
    "#group_data_true = df.groupby(['strain'],as_index=False).count()\n",
    "#group_data_true=group_data_true.rename(columns={\"length\": \"n_true\"})\n",
    "\n",
    "#group_data_true.head(7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "      <th>sensitivity</th>\n",
       "      <th>precision</th>\n",
       "      <th>specificity</th>\n",
       "      <th>f-score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>2</td>\n",
       "      <td>132</td>\n",
       "      <td>0</td>\n",
       "      <td>0.014925</td>\n",
       "      <td>0.014925</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.014925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>2</td>\n",
       "      <td>136</td>\n",
       "      <td>0</td>\n",
       "      <td>0.014493</td>\n",
       "      <td>0.014493</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.014493</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>1</td>\n",
       "      <td>145</td>\n",
       "      <td>0</td>\n",
       "      <td>0.006849</td>\n",
       "      <td>0.006849</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.006849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>2</td>\n",
       "      <td>153</td>\n",
       "      <td>0</td>\n",
       "      <td>0.012903</td>\n",
       "      <td>0.012903</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.012903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>3</td>\n",
       "      <td>169</td>\n",
       "      <td>0</td>\n",
       "      <td>0.017442</td>\n",
       "      <td>0.017442</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.017442</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTP  nFP  nTN  sensitivity  precision  \\\n",
       "0  BioGraph*         10    AKR_J    2  132    0     0.014925   0.014925   \n",
       "1  BioGraph*         10      A_J    2  136    0     0.014493   0.014493   \n",
       "2  BioGraph*         10  BALB_CJ    1  145    0     0.006849   0.006849   \n",
       "3  BioGraph*         10  C3H_HeJ    2  153    0     0.012903   0.012903   \n",
       "4  BioGraph*         10    CBA_J    3  169    0     0.017442   0.017442   \n",
       "\n",
       "   specificity   f-score  \n",
       "0          0.0  0.014925  \n",
       "1          0.0  0.014493  \n",
       "2          0.0  0.006849  \n",
       "3          0.0  0.012903  \n",
       "4          0.0  0.017442  "
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df_merge['nFN']=df_merge['n_true']-df_merge['nTP']\n",
    "\n",
    "df_merge_50_100['sensitivity']=df_merge_50_100['nTP']/(df_merge_50_100['nTP']+df_merge_50_100['nFP'])\n",
    "df_merge_50_100['precision']=df_merge_50_100['nTP']/(df_merge_50_100['nTP']+df_merge_50_100['nFP'])\n",
    "df_merge_50_100['specificity']=df_merge_50_100['nTN']/(df_merge_50_100['nTN']+df_merge_50_100['nFP'])\n",
    "df_merge_50_100['f-score']=2*(df_merge_50_100['sensitivity']*df_merge_50_100['precision'])/(df_merge_50_100['sensitivity']+df_merge_50_100['precision']+0.00000001)\n",
    "df_merge_50_100.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 100-500 bp Deletions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>114</td>\n",
       "      <td>FP</td>\n",
       "      <td>17231867</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>341</td>\n",
       "      <td>FP</td>\n",
       "      <td>18926467</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>106</td>\n",
       "      <td>FP</td>\n",
       "      <td>21474388</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>349</td>\n",
       "      <td>FP</td>\n",
       "      <td>23059354</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>470</td>\n",
       "      <td>FP</td>\n",
       "      <td>23214172</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool strain length flag  position threshold\n",
       "3   sniffles   LP_J    114   FP  17231867        10\n",
       "11  sniffles   LP_J    341   FP  18926467        10\n",
       "19  sniffles   LP_J    106   FP  21474388        10\n",
       "25  sniffles   LP_J    349   FP  23059354        10\n",
       "29  sniffles   LP_J    470   FP  23214172        10"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cat_100_500=df_cat.loc[(df_cat['length'] >= 100) & (df_cat['length'] <500)]\n",
    "df_cat_100_500.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTP\n",
       "0  BioGraph*         10    AKR_J   19\n",
       "1  BioGraph*         10      A_J   29\n",
       "2  BioGraph*         10  BALB_CJ   26\n",
       "3  BioGraph*         10  C3H_HeJ   24\n",
       "4  BioGraph*         10    CBA_J   28"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_TP_100_500 = df_cat_100_500[df_cat_100_500['flag'] == 'TP'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_TP_100_500 = group_TP_100_500.rename(columns={\"flag\": \"nTP\"})\n",
    "group_TP_100_500.shape\n",
    "group_TP_100_500.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nFP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nFP\n",
       "0  BioGraph*         10    AKR_J  153\n",
       "1  BioGraph*         10      A_J  162\n",
       "2  BioGraph*         10  BALB_CJ  151\n",
       "3  BioGraph*         10  C3H_HeJ  156\n",
       "4  BioGraph*         10    CBA_J  169"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_FP_100_500=df_cat_100_500[df_cat_100_500['flag'] == 'FP'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_FP_100_500=group_FP_100_500.rename(columns={\"flag\": \"nFP\"})\n",
    "group_FP_100_500.shape\n",
    "group_FP_100_500.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>A_J</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTN\n",
       "0  BioGraph*      10000    AKR_J    1\n",
       "1  BioGraph*      10000      A_J    1\n",
       "2  BioGraph*      10000  BALB_CJ    1\n",
       "3  BioGraph*      10000  C3H_HeJ    1\n",
       "4  BioGraph*      10000    CBA_J    1"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_TN_100_500=df_cat_100_500[df_cat_100_500['flag'] == 'TN'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_TN_100_500=group_TN_100_500.rename(columns={\"flag\": \"nTN\"})\n",
    "group_TN_100_500.shape\n",
    "group_TN_100_500.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 4)"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_TP_100_500['tool'] == t) & (group_TP_100_500['strain'] == s) & (group_TP_100_500['threshold'] == th)).any()):\n",
    "                group_TP_100_500 = group_TP_100_500.append({'tool' : t , 'strain' : s,'threshold' : th,'nTP' : 0} , ignore_index=True)\n",
    "group_TP_100_500.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 4)"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_FP_100_500['tool'] == t) & (group_FP_100_500['strain'] == s) & (group_FP_100_500['threshold'] == th)).any()):\n",
    "                group_FP_100_500 = group_FP_100_500.append({'tool' : t , 'strain' : s,'threshold' : th,'nFP' : 0} , ignore_index=True)\n",
    "group_FP_100_500.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 4)"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_TN_100_500['tool'] == t) & (group_TN_100_500['strain'] == s) & (group_TN_100_500['threshold'] == th)).any()):\n",
    "                group_TN_100_500 = group_TN_100_500.append({'tool' : t , 'strain' : s,'threshold' : th,'nTN' : 0} , ignore_index=True)\n",
    "group_TN_100_500.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 6)"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge_100_500=pd.merge(group_TP_100_500, group_FP_100_500,on=['tool','threshold','strain'])\n",
    "df_merge_100_500=pd.merge(df_merge_100_500, group_TN_100_500,on=['tool','threshold','strain'])\n",
    "df_merge_100_500.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>19</td>\n",
       "      <td>153</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>29</td>\n",
       "      <td>162</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>26</td>\n",
       "      <td>151</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>24</td>\n",
       "      <td>156</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>28</td>\n",
       "      <td>169</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>28</td>\n",
       "      <td>173</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>19</td>\n",
       "      <td>118</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>131</td>\n",
       "      <td>41</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>A_J</td>\n",
       "      <td>147</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>140</td>\n",
       "      <td>37</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTP  nFP  nTN\n",
       "0  BioGraph*         10    AKR_J   19  153    0\n",
       "1  BioGraph*         10      A_J   29  162    0\n",
       "2  BioGraph*         10  BALB_CJ   26  151    0\n",
       "3  BioGraph*         10  C3H_HeJ   24  156    0\n",
       "4  BioGraph*         10    CBA_J   28  169    0\n",
       "5  BioGraph*         10   DBA_2J   28  173    0\n",
       "6  BioGraph*         10     LP_J   19  118    0\n",
       "7  BioGraph*        100    AKR_J  131   41    0\n",
       "8  BioGraph*        100      A_J  147   44    0\n",
       "9  BioGraph*        100  BALB_CJ  140   37    0"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge_100_500.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "      <th>sensitivity</th>\n",
       "      <th>precision</th>\n",
       "      <th>specificity</th>\n",
       "      <th>f-score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>19</td>\n",
       "      <td>153</td>\n",
       "      <td>0</td>\n",
       "      <td>0.110465</td>\n",
       "      <td>0.110465</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.110465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>29</td>\n",
       "      <td>162</td>\n",
       "      <td>0</td>\n",
       "      <td>0.151832</td>\n",
       "      <td>0.151832</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.151832</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>26</td>\n",
       "      <td>151</td>\n",
       "      <td>0</td>\n",
       "      <td>0.146893</td>\n",
       "      <td>0.146893</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.146893</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>24</td>\n",
       "      <td>156</td>\n",
       "      <td>0</td>\n",
       "      <td>0.133333</td>\n",
       "      <td>0.133333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.133333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>28</td>\n",
       "      <td>169</td>\n",
       "      <td>0</td>\n",
       "      <td>0.142132</td>\n",
       "      <td>0.142132</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.142132</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTP  nFP  nTN  sensitivity  precision  \\\n",
       "0  BioGraph*         10    AKR_J   19  153    0     0.110465   0.110465   \n",
       "1  BioGraph*         10      A_J   29  162    0     0.151832   0.151832   \n",
       "2  BioGraph*         10  BALB_CJ   26  151    0     0.146893   0.146893   \n",
       "3  BioGraph*         10  C3H_HeJ   24  156    0     0.133333   0.133333   \n",
       "4  BioGraph*         10    CBA_J   28  169    0     0.142132   0.142132   \n",
       "\n",
       "   specificity   f-score  \n",
       "0          0.0  0.110465  \n",
       "1          0.0  0.151832  \n",
       "2          0.0  0.146893  \n",
       "3          0.0  0.133333  \n",
       "4          0.0  0.142132  "
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge_100_500['sensitivity']=df_merge_100_500['nTP']/(df_merge_100_500['nTP']+df_merge_100_500['nFP'])\n",
    "df_merge_100_500['precision']=df_merge_100_500['nTP']/(df_merge_100_500['nTP']+df_merge_100_500['nFP'])\n",
    "df_merge_100_500['specificity']=df_merge_100_500['nTN']/(df_merge_100_500['nTN']+df_merge_100_500['nFP'])\n",
    "df_merge_100_500['f-score']=2*(df_merge_100_500['sensitivity']*df_merge_100_500['precision'])/(df_merge_100_500['sensitivity']+df_merge_100_500['precision']+0.00000001)\n",
    "df_merge_100_500.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 500-1000 bp Deletions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>618</td>\n",
       "      <td>FP</td>\n",
       "      <td>18396654</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>709</td>\n",
       "      <td>FP</td>\n",
       "      <td>25931984</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>51</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>524</td>\n",
       "      <td>FP</td>\n",
       "      <td>31997429</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>53</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>777</td>\n",
       "      <td>FP</td>\n",
       "      <td>36762558</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>790</td>\n",
       "      <td>FP</td>\n",
       "      <td>43239099</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool strain length flag  position threshold\n",
       "9   sniffles   LP_J    618   FP  18396654        10\n",
       "33  sniffles   LP_J    709   FP  25931984        10\n",
       "51  sniffles   LP_J    524   FP  31997429        10\n",
       "53  sniffles   LP_J    777   FP  36762558        10\n",
       "74  sniffles   LP_J    790   FP  43239099        10"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cat_500_1000=df_cat.loc[(df_cat['length'] >= 500) & (df_cat['length'] <1000)]\n",
    "df_cat_500_1000.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTP\n",
       "0  BioGraph*         10    AKR_J    5\n",
       "1  BioGraph*         10      A_J    3\n",
       "2  BioGraph*         10  BALB_CJ    4\n",
       "3  BioGraph*         10  C3H_HeJ    4\n",
       "4  BioGraph*         10    CBA_J    4"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_TP_500_1000 = df_cat_500_1000[df_cat_500_1000['flag'] == 'TP'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_TP_500_1000 = group_TP_500_1000.rename(columns={\"flag\": \"nTP\"})\n",
    "group_TP_500_1000.shape\n",
    "group_TP_500_1000.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nFP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nFP\n",
       "0  BioGraph*         10    AKR_J   27\n",
       "1  BioGraph*         10      A_J   29\n",
       "2  BioGraph*         10  BALB_CJ   26\n",
       "3  BioGraph*         10  C3H_HeJ   26\n",
       "4  BioGraph*         10    CBA_J   25"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_FP_500_1000=df_cat_500_1000[df_cat_500_1000['flag'] == 'FP'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_FP_500_1000=group_FP_500_1000.rename(columns={\"flag\": \"nFP\"})\n",
    "group_FP_500_1000.shape\n",
    "group_FP_500_1000.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>A_J</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTN\n",
       "0  BioGraph*        100    AKR_J    3\n",
       "1  BioGraph*        100      A_J    2\n",
       "2  BioGraph*        100  BALB_CJ    2\n",
       "3  BioGraph*        100  C3H_HeJ    3\n",
       "4  BioGraph*        100    CBA_J    3"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_TN_500_1000=df_cat_500_1000[df_cat_500_1000['flag'] == 'TN'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_TN_500_1000=group_TN_500_1000.rename(columns={\"flag\": \"nTN\"})\n",
    "group_TN_500_1000.shape\n",
    "group_TN_500_1000.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 4)"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_TP_500_1000['tool'] == t) & (group_TP_500_1000['strain'] == s) & (group_TP_500_1000['threshold'] == th)).any()):\n",
    "                group_TP_500_1000 = group_TP_500_1000.append({'tool' : t , 'strain' : s,'threshold' : th,'nTP' : 0} , ignore_index=True)\n",
    "group_TP_500_1000.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 4)"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_FP_500_1000['tool'] == t) & (group_FP_500_1000['strain'] == s) & (group_FP_500_1000['threshold'] == th)).any()):\n",
    "                group_FP_500_1000 = group_FP_500_1000.append({'tool' : t , 'strain' : s,'threshold' : th,'nFP' : 0} , ignore_index=True)\n",
    "group_FP_500_1000.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 4)"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_TN_500_1000['tool'] == t) & (group_TN_500_1000['strain'] == s) & (group_TN_500_1000['threshold'] == th)).any()):\n",
    "                group_TN_500_1000 = group_TN_500_1000.append({'tool' : t , 'strain' : s,'threshold' : th,'nTN' : 0} , ignore_index=True)\n",
    "group_TN_500_1000.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 6)"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge_500_1000=pd.merge(group_TP_500_1000, group_FP_500_1000,on=['tool','threshold','strain'])\n",
    "df_merge_500_1000=pd.merge(df_merge_500_1000, group_TN_500_1000,on=['tool','threshold','strain'])\n",
    "df_merge_500_1000.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>5</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>3</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>4</td>\n",
       "      <td>26</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>4</td>\n",
       "      <td>26</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>4</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>3</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>28</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>A_J</td>\n",
       "      <td>26</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTP  nFP  nTN\n",
       "0  BioGraph*         10    AKR_J    5   27    0\n",
       "1  BioGraph*         10      A_J    3   29    0\n",
       "2  BioGraph*         10  BALB_CJ    4   26    0\n",
       "3  BioGraph*         10  C3H_HeJ    4   26    0\n",
       "4  BioGraph*         10    CBA_J    4   25    0\n",
       "5  BioGraph*         10   DBA_2J    3   27    0\n",
       "6  BioGraph*         10     LP_J    3   19    0\n",
       "7  BioGraph*        100    AKR_J   28    4    3\n",
       "8  BioGraph*        100      A_J   26    6    2\n",
       "9  BioGraph*        100  BALB_CJ   25    5    2"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge_500_1000.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "      <th>sensitivity</th>\n",
       "      <th>precision</th>\n",
       "      <th>specificity</th>\n",
       "      <th>f-score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>5</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>0.156250</td>\n",
       "      <td>0.156250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.156250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>3</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>0.093750</td>\n",
       "      <td>0.093750</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.093750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>4</td>\n",
       "      <td>26</td>\n",
       "      <td>0</td>\n",
       "      <td>0.133333</td>\n",
       "      <td>0.133333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.133333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>4</td>\n",
       "      <td>26</td>\n",
       "      <td>0</td>\n",
       "      <td>0.133333</td>\n",
       "      <td>0.133333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.133333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>4</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>0.137931</td>\n",
       "      <td>0.137931</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.137931</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTP  nFP  nTN  sensitivity  precision  \\\n",
       "0  BioGraph*         10    AKR_J    5   27    0     0.156250   0.156250   \n",
       "1  BioGraph*         10      A_J    3   29    0     0.093750   0.093750   \n",
       "2  BioGraph*         10  BALB_CJ    4   26    0     0.133333   0.133333   \n",
       "3  BioGraph*         10  C3H_HeJ    4   26    0     0.133333   0.133333   \n",
       "4  BioGraph*         10    CBA_J    4   25    0     0.137931   0.137931   \n",
       "\n",
       "   specificity   f-score  \n",
       "0          0.0  0.156250  \n",
       "1          0.0  0.093750  \n",
       "2          0.0  0.133333  \n",
       "3          0.0  0.133333  \n",
       "4          0.0  0.137931  "
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge_500_1000['sensitivity']=df_merge_500_1000['nTP']/(df_merge_500_1000['nTP']+df_merge_500_1000['nFP'])\n",
    "df_merge_500_1000['precision']=df_merge_500_1000['nTP']/(df_merge_500_1000['nTP']+df_merge_500_1000['nFP'])\n",
    "df_merge_500_1000['specificity']=df_merge_500_1000['nTN']/(df_merge_500_1000['nTN']+df_merge_500_1000['nFP'])\n",
    "df_merge_500_1000['f-score']=2*(df_merge_500_1000['sensitivity']*df_merge_500_1000['precision'])/(df_merge_500_1000['sensitivity']+df_merge_500_1000['precision']+0.00000001)\n",
    "df_merge_500_1000.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1000+ bp Deletions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1725</td>\n",
       "      <td>FP</td>\n",
       "      <td>6055203</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>34582728</td>\n",
       "      <td>FP</td>\n",
       "      <td>17098523</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>2168</td>\n",
       "      <td>FP</td>\n",
       "      <td>18195112</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>4150</td>\n",
       "      <td>FP</td>\n",
       "      <td>18920153</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1148</td>\n",
       "      <td>FP</td>\n",
       "      <td>21448557</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool strain    length flag  position threshold\n",
       "0   sniffles   LP_J      1725   FP   6055203        10\n",
       "1   sniffles   LP_J  34582728   FP  17098523        10\n",
       "7   sniffles   LP_J      2168   FP  18195112        10\n",
       "10  sniffles   LP_J      4150   FP  18920153        10\n",
       "18  sniffles   LP_J      1148   FP  21448557        10"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cat_1000=df_cat.loc[(df_cat['length'] >= 1000)]\n",
    "df_cat_1000.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTP\n",
       "0  BioGraph*         10    AKR_J   10\n",
       "1  BioGraph*         10      A_J   13\n",
       "2  BioGraph*         10  BALB_CJ   14\n",
       "3  BioGraph*         10  C3H_HeJ   11\n",
       "4  BioGraph*         10    CBA_J    9"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_TP_1000 = df_cat_1000[df_cat_1000['flag'] == 'TP'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_TP_1000 = group_TP_1000.rename(columns={\"flag\": \"nTP\"})\n",
    "group_TP_1000.shape\n",
    "group_TP_1000.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nFP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nFP\n",
       "0  BioGraph*         10    AKR_J   77\n",
       "1  BioGraph*         10      A_J   94\n",
       "2  BioGraph*         10  BALB_CJ   95\n",
       "3  BioGraph*         10  C3H_HeJ   81\n",
       "4  BioGraph*         10    CBA_J   82"
      ]
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_FP_1000 = df_cat_1000[df_cat_1000['flag'] == 'FP'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_FP_1000 = group_FP_1000.rename(columns={\"flag\": \"nFP\"})\n",
    "group_FP_1000.shape\n",
    "group_FP_1000.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTN\n",
       "0  BioGraph*         10    AKR_J   11\n",
       "1  BioGraph*         10      A_J   20\n",
       "2  BioGraph*         10  BALB_CJ   13\n",
       "3  BioGraph*         10  C3H_HeJ   17\n",
       "4  BioGraph*         10    CBA_J   18"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_TN_1000 = df_cat_1000[df_cat_1000['flag'] == 'TN'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_TN_1000 = group_TN_1000.rename(columns={\"flag\": \"nTN\"})\n",
    "group_TN_1000.shape\n",
    "group_TN_1000.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 4)"
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_TP_1000['tool'] == t) & (group_TP_1000['strain'] == s) & (group_TP_1000['threshold'] == th)).any()):\n",
    "                group_TP_1000 = group_TP_1000.append({'tool' : t , 'strain' : s,'threshold' : th,'nTP' : 0} , ignore_index=True)\n",
    "group_TP_1000.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 4)"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_FP_1000['tool'] == t) & (group_FP_1000['strain'] == s) & (group_FP_1000['threshold'] == th)).any()):\n",
    "                group_FP_1000 = group_FP_1000.append({'tool' : t , 'strain' : s,'threshold' : th,'nFP' : 0} , ignore_index=True)\n",
    "group_FP_1000.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 4)"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_TN_1000['tool'] == t) & (group_TN_1000['strain'] == s) & (group_TN_1000['threshold'] == th)).any()):\n",
    "                group_TN_1000 = group_TN_1000.append({'tool' : t , 'strain' : s,'threshold' : th,'nTN' : 0} , ignore_index=True)\n",
    "group_TN_1000.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 6)"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge_1000=pd.merge(group_TP_1000, group_FP_1000,on=['tool','threshold','strain'])\n",
    "df_merge_1000=pd.merge(df_merge_1000, group_TN_1000,on=['tool','threshold','strain'])\n",
    "df_merge_1000.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "      <th>sensitivity</th>\n",
       "      <th>precision</th>\n",
       "      <th>specificity</th>\n",
       "      <th>f-score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>5</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>0.156250</td>\n",
       "      <td>0.156250</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.156250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>3</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>0.093750</td>\n",
       "      <td>0.093750</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.093750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>4</td>\n",
       "      <td>26</td>\n",
       "      <td>0</td>\n",
       "      <td>0.133333</td>\n",
       "      <td>0.133333</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.133333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>4</td>\n",
       "      <td>26</td>\n",
       "      <td>0</td>\n",
       "      <td>0.133333</td>\n",
       "      <td>0.133333</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.133333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>4</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>0.137931</td>\n",
       "      <td>0.137931</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.137931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>3</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.100000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>0.136364</td>\n",
       "      <td>0.136364</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.136364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>28</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>0.875000</td>\n",
       "      <td>0.875000</td>\n",
       "      <td>0.428571</td>\n",
       "      <td>0.875000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>A_J</td>\n",
       "      <td>26</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>0.812500</td>\n",
       "      <td>0.812500</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>0.812500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>0.833333</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTP  nFP  nTN  sensitivity  precision  \\\n",
       "0  BioGraph*         10    AKR_J    5   27    0     0.156250   0.156250   \n",
       "1  BioGraph*         10      A_J    3   29    0     0.093750   0.093750   \n",
       "2  BioGraph*         10  BALB_CJ    4   26    0     0.133333   0.133333   \n",
       "3  BioGraph*         10  C3H_HeJ    4   26    0     0.133333   0.133333   \n",
       "4  BioGraph*         10    CBA_J    4   25    0     0.137931   0.137931   \n",
       "5  BioGraph*         10   DBA_2J    3   27    0     0.100000   0.100000   \n",
       "6  BioGraph*         10     LP_J    3   19    0     0.136364   0.136364   \n",
       "7  BioGraph*        100    AKR_J   28    4    3     0.875000   0.875000   \n",
       "8  BioGraph*        100      A_J   26    6    2     0.812500   0.812500   \n",
       "9  BioGraph*        100  BALB_CJ   25    5    2     0.833333   0.833333   \n",
       "\n",
       "   specificity   f-score  \n",
       "0     0.000000  0.156250  \n",
       "1     0.000000  0.093750  \n",
       "2     0.000000  0.133333  \n",
       "3     0.000000  0.133333  \n",
       "4     0.000000  0.137931  \n",
       "5     0.000000  0.100000  \n",
       "6     0.000000  0.136364  \n",
       "7     0.428571  0.875000  \n",
       "8     0.250000  0.812500  \n",
       "9     0.285714  0.833333  "
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge_500_1000.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "      <th>sensitivity</th>\n",
       "      <th>precision</th>\n",
       "      <th>specificity</th>\n",
       "      <th>f-score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>10</td>\n",
       "      <td>77</td>\n",
       "      <td>11</td>\n",
       "      <td>0.114943</td>\n",
       "      <td>0.114943</td>\n",
       "      <td>0.125000</td>\n",
       "      <td>0.114943</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>13</td>\n",
       "      <td>94</td>\n",
       "      <td>20</td>\n",
       "      <td>0.121495</td>\n",
       "      <td>0.121495</td>\n",
       "      <td>0.175439</td>\n",
       "      <td>0.121495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>14</td>\n",
       "      <td>95</td>\n",
       "      <td>13</td>\n",
       "      <td>0.128440</td>\n",
       "      <td>0.128440</td>\n",
       "      <td>0.120370</td>\n",
       "      <td>0.128440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>11</td>\n",
       "      <td>81</td>\n",
       "      <td>17</td>\n",
       "      <td>0.119565</td>\n",
       "      <td>0.119565</td>\n",
       "      <td>0.173469</td>\n",
       "      <td>0.119565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>9</td>\n",
       "      <td>82</td>\n",
       "      <td>18</td>\n",
       "      <td>0.098901</td>\n",
       "      <td>0.098901</td>\n",
       "      <td>0.180000</td>\n",
       "      <td>0.098901</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTP  nFP  nTN  sensitivity  precision  \\\n",
       "0  BioGraph*         10    AKR_J   10   77   11     0.114943   0.114943   \n",
       "1  BioGraph*         10      A_J   13   94   20     0.121495   0.121495   \n",
       "2  BioGraph*         10  BALB_CJ   14   95   13     0.128440   0.128440   \n",
       "3  BioGraph*         10  C3H_HeJ   11   81   17     0.119565   0.119565   \n",
       "4  BioGraph*         10    CBA_J    9   82   18     0.098901   0.098901   \n",
       "\n",
       "   specificity   f-score  \n",
       "0     0.125000  0.114943  \n",
       "1     0.175439  0.121495  \n",
       "2     0.120370  0.128440  \n",
       "3     0.173469  0.119565  \n",
       "4     0.180000  0.098901  "
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge_1000['sensitivity']=df_merge_1000['nTP']/(df_merge_1000['nTP']+df_merge_1000['nFP'])\n",
    "df_merge_1000['precision']=df_merge_1000['nTP']/(df_merge_1000['nTP']+df_merge_1000['nFP'])\n",
    "df_merge_1000['specificity']=df_merge_1000['nTN']/(df_merge_1000['nTN']+df_merge_1000['nFP'])\n",
    "df_merge_1000['f-score']=2*(df_merge_1000['sensitivity']*df_merge_1000['precision'])/(df_merge_1000['sensitivity']+df_merge_1000['precision']+0.00000001)\n",
    "df_merge_1000.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
